component/http/v2/component_option.go

```diff
@@ -0,0 +1,77 @@
+package v2
+
+import (
+	"errors"
+	"time"
+)
+
+// OptionFunc definition for configuring the component in a functional way.
+type OptionFunc func(*Component) error
+
+// SSL functional option.
+func SSL(cert, key string) OptionFunc {
```

Comment:
Since Go calls it TLS, shall we also call it TLS?

Reply:
Done